semver/format.ts

```diff
@@ -0,0 +1,91 @@
+// Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.
+import { SemVerComparator } from "./comparator.ts";
+import { SemVerRange } from "./range.ts";
+import { SemVer } from "./semver.ts";
+import { FormatStyle } from "./types.ts";
+
+/**
+ * Format a SemVer object into a string.
+ *
+ * If any number is NaN then NaN will be printed.
+ *
+ * If any number is positive or negative infinity then '∞' or '⧞' will be printed instead.
+ *
+ * @param semver The semantic version to format
+ * @returns The string representation of a smenatic version.
+ */
+export function format(semver: SemVer, style: FormatStyle = "full") {
+  const major = formatNumber(semver.major);
+  const minor = formatNumber(semver.minor);
+  const patch = formatNumber(semver.patch);
+  const pre = semver.prerelease.join(".");
+  const build = semver.build.join(".");
+
+  const primary = `${major}.${minor}.${patch}`;
+  const release = [primary, pre].filter((v) => v).join("-");
+  const full = [release, build].filter((v) => v).join("+");
+  //           ┌───── full
+  //       ┌───┴───┐
+  //       ├───────── release
+  //   ┌───┴───┐   │
+  //   ├───────────── primary
+  // ┌─┴─┐     │   │
+  // 1.2.3-pre.1+b.1
+  // │ │ │ └─┬─┘ └┬┘
+  // │ │ │   │    └── build
+  // │ │ │   └─────── pre
+  // │ │ └─────────── patch
+  // │ └───────────── minor
+  // └─────────────── major
```

Comment:
This is a nice visual aid 🔍 👀

Reply:
I moved it to a formatting section in the main module documentation.